app/src/test/java/IntroScreensActivityViewModelTest.kt

```diff
@@ -0,0 +1,129 @@
+
+import `in`.projecteka.jataayu.R
+import `in`.projecteka.jataayu.presentation.IntroScreensActivityViewModel
+import `in`.projecteka.jataayu.util.repository.PreferenceRepository
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.*
+import org.mockito.MockitoAnnotations
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class IntroScreensActivityViewModelTest {
+
+    @Mock
+    private lateinit var preferenceRepository: PreferenceRepository
+
+    @get:Rule
+    val taskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var introScreensActivityViewModel: IntroScreensActivityViewModel
+
+    @Before
+    fun setup() {
+        MockitoAnnotations.initMocks(this)
+        introScreensActivityViewModel = IntroScreensActivityViewModel(preferenceRepository)
+        introScreensActivityViewModel.init()
+    }
+
+    @After
+    fun tearDown() {
+        verifyNoMoreInteractions(preferenceRepository)
+        validateMockitoUsage()
+    }
+
+    @Test
+    fun `should initialize view model with correct values`() {
+        assertEquals(4, introScreensActivityViewModel.layouts?.size)
+
+        assertEquals(0, introScreensActivityViewModel.setViewpagerCurrentItemEvent.value)
+
+        introScreensActivityViewModel.addBottomDotsEvent.observeForever{
+            assert(true)
+        }
+    }
+
+    @Test
+    fun `should select first page`(){
```

Comment:
can you give some meaningful name then first page, second page?

Reply:
"first onboarding screen" will work?